pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -369,9 +369,21 @@ func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(statusReplicas int32
 	usageRatio := float64(usage) / (float64(targetUsagePerPod) * float64(replicaCount))
 	if math.Abs(1.0-usageRatio) > c.tolerance {
 		// update number of replicas if the change is large enough
-		replicaCount = int32(math.Ceil(float64(usage) / float64(targetUsagePerPod)))
+		replicaCountResult := math.Ceil(float64(usage) / float64(targetUsagePerPod))
```

Comment:
Should we calculate here first and then use the math.Ceil method? 
like this: 
```go
replicaCountResult := float64(usage) / float64(targetUsagePerPod)
if replicaCountResult > float64(math.MaxInt32) {
   replicaCount = math.MaxInt32
} else {
   replicaCount = math.Ceil(int32(replicaCountResult))
}
```

Reply:
Yes. it makes sense. Ill adjust thanks :)